src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -524,35 +553,36 @@ private void processCloudWatchAttemptLogInformation(Map<String, Set<LogFile>> co
+                + cloudWatchAttemptLogFileInformation.getStartPosition();
+
+        if (!logFileGroup.isActiveFile(file) && file.length() == bytesUploaded) {
             Set<LogFile> completedFiles = completedLogFilePerComponent.getOrDefault(componentName,
                     new HashSet<>());
             completedFiles.add(file);
             completedLogFilePerComponent.put(componentName, completedFiles);
-            if (currentProcessingLogFilePerComponent.containsKey(componentName)) {
-                CurrentProcessingFileInformation fileInformation = currentProcessingLogFilePerComponent
-                        .get(componentName);
-                if (fileInformation.fileHash.equals(fileHash)) {
-                    currentProcessingLogFilePerComponent.remove(componentName);
-                }
-            }
-        } else {
-            // Add the file to the current processing list for the component.
-            // Note: There should always be only 1 file which will be in progress at any given time.
-            // @deprecated The fileName is deprecated value in versions greater than 2.2, but keep it here to avoid
-            // upgrade-downgrade issues.
-            CurrentProcessingFileInformation processingFileInformation =
-                    CurrentProcessingFileInformation.builder()
-                            .fileName(file.getSourcePath())
-                            .startPosition(cloudWatchAttemptLogFileInformation.getStartPosition()
-                                    + cloudWatchAttemptLogFileInformation.getBytesRead())
-                            .lastModifiedTime(cloudWatchAttemptLogFileInformation.getLastModifiedTime())
-                            .fileHash(fileHash)
-                            .build();
-            currentProcessingLogFilePerComponent.put(attemptLogInformation.getComponentName(),
-                    processingFileInformation);
         }
+
+
+        // Track each file that got uploaded. So that if modified in the future we can still track upload
+        // the new contents starting from the last upload position.
+
+        CurrentProcessingFileInformation processingFileInformation =
+                CurrentProcessingFileInformation.builder()
+                        .fileName(file.getSourcePath()) // @deprecated
+                        .startPosition(bytesUploaded)
+                        .lastModifiedTime(cloudWatchAttemptLogFileInformation.getLastModifiedTime())
+                        .fileHash(fileHash)
+                        .build();
+
+        if (!processingFilesInformation.containsKey(componentName)) {
+            // TODO: need to figure a smarter mechanism to adjust the LRU capacity. Add an adjust capacity method
+            processingFilesInformation.put(componentName, new ProcessingFileLRU(
+                    Math.max(logFileGroup.getLogFiles().size(), DEFAULT_MAX_FILES_TO_TRACK_PER_COMPONENT)));
```

Comment:
This only gives you the log files that have been modified since the last iteration, right? I think this would potentially remove hashes prematurely

Reply:
This is just initializing the LRU, in case there is nothing in there. The only time at which they would get removed is on line 585 below if we have more than 5 files. Shouldn't be a problem unless there are more than 5 active files that match the regex